plugins/devlink/fu-devlink-device.c

```diff
@@ -451,24 +418,13 @@ fu_devlink_device_info_cb(const struct nlmsghdr *nlh, gpointer data)
 		fu_device_set_serial(device, serial_number);
 	}
 
-	version_table = fu_devlink_device_get_version_table(nlh);
-
-	/* remove components that are not in the version table */
-	for (guint i = 0; i < children->len; i++) {
-		FuDevice *component = g_ptr_array_index(children, children->len - i - 1);
-		const gchar *name = fu_device_get_name(component);
-		FuDevlinkVersionInfo *version_info = g_hash_table_lookup(version_table, name);
-
```

Comment:
Not sure why you removed this code. It is possible the component gets removed over devlink reload, we need to treat that correctly.